### PR TITLE
Feature/add updater version

### PR DESCRIPTION
**Pull request by pvyawaha**
*Description of changes:*
The change adds the updater version to the job status. Updater version is the version of the firmware image from which the new version is updated.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/ota_mqtt.c [hunk 1]
```diff
@@ -209,13 +209,15 @@ static uint32_t prvBuildStatusMessageSelfTest( char * pMsgBuffer,
...
+ * @param[in] updaterVersion Version from which the new version was updated.
  * @return uint32_t Size of the message.
  */
 static uint32_t prvBuildStatusMessageFinish( char * pMsgBuffer,
                                              size_t msgBufferSize,
                                              OtaJobStatus_t status,
                                              int32_t reason,
-                                             int32_t subReason );
+                                             int32_t subReason,
+                                             uint32_t updaterVersion );
```
Comment (aggarg) [inline source/ota_mqtt.c:220]:
```suggestion
                                             uint32_t previousVersion );
```

File: source/ota_mqtt.c [hunk 2]
```diff
@@ -725,6 +732,13 @@ static uint32_t prvBuildStatusMessageFinish( char * pMsgBuffer,
         NULL, /* Version minor string not available at compile time, initialized below. */
         ".",
         NULL, /* Version build string not available at compile time, initialized below. */
+        "\",\"" OTA_JSON_UPDATED_BY_KEY_ONLY "\":\"",
+        "  v",
+        NULL, /* Version major string not available at compile time, initialized below. */
+        ".",
+        NULL, /* Version minor string not available at compile time, initialized below. */
+        ".",
+        NULL, /* Version build string not available at compile time, initialized below. */
```
Comment (aggarg) [inline source/ota_mqtt.c:741]:
```suggestion
        NULL, /* Previous version major string not available at compile time, initialized below. */
        ".",
        NULL, /* Previous version minor string not available at compile time, initialized below. */
        ".",
        NULL, /* Previous version build string not available at compile time, initialized below. */
```

File: source/ota_mqtt.c [hunk 3]
```diff
@@ -725,6 +732,13 @@ static uint32_t prvBuildStatusMessageFinish( char * pMsgBuffer,
         NULL, /* Version minor string not available at compile time, initialized below. */
         ".",
         NULL, /* Version build string not available at compile time, initialized below. */
+        "\",\"" OTA_JSON_UPDATED_BY_KEY_ONLY "\":\"",
```
Comment (divekarshubham) [inline source/ota_mqtt.c:735]:
```suggestion
        "\",\"" OTA_JSON_UPDATED_BY_KEY_ONLY "\":\"", /* Expands to `","updatedBy":` */
```

File: source/ota_mqtt.c [hunk 4]
```diff
@@ -687,15 +689,20 @@ static uint32_t prvBuildStatusMessageFinish( char * pMsgBuffer,
                                              size_t msgBufferSize,
                                              OtaJobStatus_t status,
                                              int32_t reason,
-                                             int32_t subReason )
+                                             int32_t subReason,
+                                             uint32_t updaterVersion )
```
Comment (divekarshubham) [inline source/ota_mqtt.c:693]: Inconsistent with the above declaration ie `previousVersion`

Comment (pvyawaha) [reply source/ota_mqtt.c]: https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/232/commits/fcd5a6e66b056dbed4623bdf68f9bb90e061843c

File: source/ota_mqtt.c [hunk 5]
```diff
@@ -687,15 +689,20 @@ static uint32_t prvBuildStatusMessageFinish( char * pMsgBuffer,
...
 {
     uint32_t msgSize = 0;
     char reasonString[ U32_MAX_LEN + 1 ];
     char subReasonString[ U32_MAX_LEN + 1 ];
     char versionMajorString[ U32_MAX_LEN + 1 ];
```
[the remark this answers is not in the document]
Comment (pvyawaha) [reply source/ota_mqtt.c]: https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/232/commits/fcd5a6e66b056dbed4623bdf68f9bb90e061843c

**Review by aggarg — APPROVED** [review]

**Review by divekarshubham — APPROVED** [review]

**Pull request merged**